### PR TITLE
refactor: factor test result saving to function

### PR DESCRIPTION
**Pull request by tylergu**

### CODE REVIEW

File: sieve.py [hunk 1]
```diff
@@ -26,6 +26,56 @@
     sieve_stages,
 )
 
+def save_run_result(project, test, mode, stage, test_config, alarm, bug_report, starttime):
+    if stage is not sieve_stages.TEST or mode is sieve_modes.VANILLA:
```
Comment (marshtompsxd) [inline sieve.py:30]: I am not sure about using `is` here. `is` returns True is the two are the same object, so I suspect the `is` here will always return False. We probably should use `==` to be safe.

Comment (tylergu) [reply sieve.py:30]: Oh I thought these are used interchangeably. Nice catch, glad it didn't turn into a wired bug.

Comment (tylergu) [reply sieve.py:30]: I think we are comparing strings here, so it should be fine. But I will change it anyway.

Comment (marshtompsxd) [reply sieve.py:30]: `sieve_stages.TEST` is a constant string but `stage` is not -- during runtime they two will not be the same object. Check this: https://stackoverflow.com/questions/1504717/why-does-comparing-strings-using-either-or-is-sometimes-produce-a-differe

**Pull request merged**